operators/pkg/instance-controller/status_inspection.go

```diff
@@ -15,6 +16,43 @@ import (
 	crownlabsv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
 )
 
+func (r *InstanceReconciler) getContainerStatus(ctx context.Context,
```

Comment:
Why isn't this function based only on the readiness status of the deployment? 

BTW, the best thing would be avoid the loop/sleep approach and return from the reconcile, watching for changes in the deployments. When the deployment becomes ready, the controller awakes, performs a new reconcile and sets the instance ready.

This change would make no longer work the bootTime metric, but I do not feel it a problem for the moment.